### PR TITLE
Add pyright + __status__ endpoint

### PR DESCRIPTION
**Pull request by Peddle**
# What is this?
Adds pyright to cicd + updates existing code to pass pyright. Also adds __status__ endpoint

# Why?
improved code quality / readability / reduces bugs in future.

__status__ endpoint is a handy healthcheck and can be extended with GPU status in future

# How did you test it works without  regressions?
Follow on PR to add unit testing + integration tests (should block master release on this). For now simple manual tests seem to pass

# If this is a new feature what may a critical error look like? 

### Things to consider to not repeat mistakes we've learned from many times
- [ ] If critical errors fire do we ping team in some obvious way (e.g., slack)? 
- [ ] Are there debug logs + a way to see these logs if we need to debug?
- [ ] Is this documented enough a dev could work on this code without getting stuck or having to ping you?

### CODE REVIEW

File: potassium/potassium.py [hunk 1]
```diff
@@ -199,7 +203,15 @@ def handle(path):
                 abort(404)
 
             endpoint = self._endpoints[route]
-            return self._handle_generic(route, endpoint, request)
+            return self._handle_generic(endpoint, request)
+
+        # add __status__ endpoint
+        @flask_app.route('/__status__', methods=["GET"])
```
Comment (ErikKaum) [inline potassium/potassium.py:209]: I think it's good to communicate elsewhere, that if you call this when the app is deployed on banana, you'll get billed for GPU time.

**Comment by ErikKaum** [comment]
All in all really legit 🔥
- one note on making the Exception have an error message (not blocking but suggesting)
- when the __status__ endpoint is called on banana infra, it allocates a GPU & bills for it (more just checking that you're aware of this)
- really nice with pyright 💪
- **NOTE**: remember to bump version before you deploy

File: potassium/potassium.py [hunk 2]
```diff
@@ -24,6 +23,9 @@ def __init__(self, status: int = 200, json: dict = {}):
         self.json = json
         self.status = status
 
+class InvalidEndpointTypeException(Exception):
+    pass
```
[the remark this answers is not in the document]
Comment (Peddle) [reply potassium/potassium.py]: done

File: potassium/potassium.py (hunk 1 above)
Comment (Peddle) [reply potassium/potassium.py:208]: discussed offline but i think this is ok given we only forward POSTs through proxy so even if a user tries to use this they won't get billed.

**Review by erik-dunteman — APPROVED** [review]
LGTM

**Pull request merged**